ibl_pipeline/analyses/behavior.py

```diff
@@ -1,81 +1,191 @@
-            key_psy['rt'] = np.hstack([rt_stim_left, rt_no_contrast, rt_stim_right])
+        if has_right:
+            contrasts_right, n_trials_stim_right = (Contrasts & key).fetch1(
+                'contrasts_right', 'n_trials_stim_right')
+            n_right_trials_stim_right = [len(trials & 'ABS(trial_stim_contrast_right-{})<1e-10'.format(contrast) & \
+                                            'trial_response_choice="CCW"') \
+                                        for contrast in contrasts_right]
+            p_right_stim_right = np.divide(n_right_trials_stim_right, n_trials_stim_right)
         else:
-            key_psy['contrasts'] = np.hstack([np.negative(contrasts_left[::-1]),  contrasts_right])*100
-            key_psy['prob_choose_right'] = np.hstack([(p_right_stim_left[::-1]), p_right_stim_right])
-            n_total_trials = np.hstack([n_trials_stim_left, n_trials_stim_right])
-            key_psy['rt'] = np.hstack([rt_stim_left, rt_stim_right])
+            p_right_stim_right = []
+        
+        if has_no_contrast:
+            n_trials_stim_no_contrast = (Contrasts & key).fetch1(
+                'n_trials_stim_no_contrast')
+            trials_no_contrast = trials & 'trial_stim_contrast_right=0' & 'trial_stim_contrast_left=0'
+            p_right_no_contrast = len(trials_no_contrast & 'trial_response_choice="CCW"')/n_trials_stim_no_contrast        
+        else:
+            p_right_no_contrast = []
+        
+        prob_choose_right = np.hstack([p_right_stim_left, p_right_no_contrast, p_right_stim_right])
 
-        pars, L = psy.mle_fit_psycho(np.vstack([key_psy['contrasts'], n_total_trials, key_psy['prob_choose_right']]), \
+        pars, L = psy.mle_fit_psycho(np.vstack([contrasts, n_trials_stim, prob_choose_right]), \
                 P_model='erf_psycho_2gammas', \
-                parstart=np.array([key_psy['contrasts'].mean(), 20., 0.05, 0.05]), \
-                parmin=np.array([key_psy['contrasts'].mean(), 0., 0., 0.]), \
-                parmax=np.array([key_psy['contrasts'].mean(), 100., 1, 1]))
-        print(pars)
+                parstart=np.array([contrasts.mean(), 20., 0.05, 0.05]), \
+                parmin=np.array([contrasts.mean(), 0., 0., 0.]), \
+                parmax=np.array([contrasts.mean(), 100., 1, 1]))
+        
+        
+
+        key_psy['prob_choose_right'] = prob_choose_right
         key_psy['bias'] = pars[0]
         key_psy['threshold'] = pars[1] 
         key_psy['lapse_low'] = pars[2]
         key_psy['lapse_high'] = pars[3]
 
         self.insert1(key_psy)
-        
\ No newline at end of file
+
+
+@schema
+class ReactionTime(dj.Computed):
+    definition = """
+    -> Contrasts
+    ---
+    reaction_time:     blob   # reaction time for all contrasts
+    """
+
+    key_source = behavior.TrialSet & \
+        (behavior.CompleteTrialSession & 'stim_on_times_status != "Missing"') & \
+        PsychResults
+
+    def make(self, key):
+        
+        key_rt = key.copy()
```

Comment:
please choose a better name than `key_rt`. Use full words. Use `key` only for primary keys.